src/status_im/ui/components/chat_icon/screen.cljs

```diff
@@ -27,16 +28,18 @@
        [react/view pending-outer-circle
         [react/view pending-inner-circle]]])))
 
-(defview chat-icon-view [chat-id _group-chat name _online styles & [hide-dapp?]]
-  (letsubs [photo-path [:contacts/chat-photo chat-id]
-            dapp?      [:get-in [:contacts/contacts chat-id :dapp?]]]
-    [react/view (:container styles)
-     (if-not (string/blank? photo-path)
-       [photos/photo photo-path styles]
-       [default-chat-icon name styles])
-     (when (and dapp? (not hide-dapp?))
-       [dapp-badge styles])
-     [pending-contact-badge chat-id styles]]))
+(defn chat-icon-view
+  [chat-id _group-chat name _online styles & [hide-dapp?]]
+  (fn [chat-id _group-chat name _online styles & [hide-dapp?]]
+    (let [photo-path (re-frame.core/subscribe [:contacts/chat-photo chat-id])
```

Comment:
assuming form-3 is not a thing anymore and subs duplication is not an issue we can rewrite `defview` and remove it later

Reply:
you mean remove defview everywhere? I'm not against it but I suppose you don't mean in this particular PR :D

Reply:
that can be done, I mean removal of defview, but I rather mean if you make it form1 in defview this subs related issue should be fixed in other places. Well better not to do this, because some logic might be based on form-3 behavior...